code/game/objects/items/weapons/shards.dm

```diff
@@ -48,80 +54,66 @@
 			if(affecting.is_robotic())
 				return
 			to_chat(H, "<span class='warning'>[src] cuts into your hand!</span>")
-			if(affecting.receive_damage(force*0.5))
+			if(affecting.receive_damage(force * 0.5))
 				H.UpdateDamageIcon()
 
 /obj/item/shard/attackby(obj/item/I, mob/user, params)
 	if(iswelder(I))
 		var/obj/item/weldingtool/WT = I
 		if(WT.remove_fuel(0, user))
-			var/obj/item/stack/sheet/glass/NG = new (user.loc)
-			for(var/obj/item/stack/sheet/glass/G in user.loc)
+			var/obj/item/stack/sheet/NG = new welded_type(user.loc)
+			for(var/obj/item/stack/sheet/G in user.loc)
+				if(!istype(G, welded_type))
+					continue
 				if(G == NG)
 					continue
 				if(G.amount >= G.max_amount)
 					continue
 				G.attackby(NG, user)
 			to_chat(user, "<span class='notice'>You add the newly-formed glass to the stack. It now contains [NG.amount] sheet\s.</span>")
 			qdel(src)
-	else
-		return ..()
+		return
+	if(istype(I, /obj/item/lightreplacer))
+		I.attackby(src, user)
+		return
+	return ..()
 
-/obj/item/shard/Crossed(AM as mob|obj, oldloc)
-	if(isliving(AM))
-		var/mob/living/M = AM
-		if(M.incorporeal_move || M.flying || M.throwing)//you are incorporal or flying or being thrown ..no shard stepping!
+/obj/item/shard/Crossed(mob/living/L, oldloc)
+	if(istype(L) && has_gravity(loc))
```

Comment:
this gravity check should prevent the original problem of getting wrecked, in space, by glass while still allowing the *tactical shoe steal and throw* strat.